General code update

dzenanz opened the pull request:
This does not work with Python 3.12, as ITK 5.3 is not compiled for it.
Python 3.11 or earlier is required.

dzenanz committed:
--pre is required to install alpha version of itkwidgets

This does not work with Python 3.12, as ITK 5.3 is not compiled for it.
Python 3.11 or earlier is required.

dzenanz committed:
Avoid ModuleNotFoundError: No module named 'zarr.storage'

dzenanz committed:
WIP: running locally

thewtex commented on notebooks/ProbabilityMapThicknessORMIR.ipynb:
```diff
@@ -785,7 +785,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.8.19"
```
This should stay at 3.8, which has been tested for the ML deps.

dzenanz committed:
WIP: Turn FullDemo.ipynb into a script form (pipeline skeleton)

This is more suitable for debugging, and is a step towards full automation

dzenanz committed:
BUG: make it run

dzenanz committed:
STYLE: Remove unused function cast_image_as

dzenanz committed:
BUG: Compute thickness before transforming to atlas space

dzenanz committed:
WIP: mesh is a vtkPolyData, transform is an ITK class

dzenanz committed:
ENH: Convert VTK mesh into ITK format and vice versa via files

These intermediate files can also be used for inspection later.

dzenanz committed:
ENH: Allow keeping intermediate outputs for inspection

dzenanz committed:
STYLE: remove redundant image saving operation

dzenanz committed:
Use uniGradICON instead of OAI_knees_registration_model (ITK 5.4)

dzenanz committed:
ENH: Orient all images into RAS to match the atlas

dzenanz committed:
ENH: Pre-process the image intensities upon reading

dzenanz committed:
ENH: Support reading .ply meshes

dzenanz committed:
ENH: Atlas and subject thickness computation are the same

dzenanz committed:
ENH: Segmentation works correctly only in DICOM PIR orientation

dzenanz committed:
ENH: Use manually crafted atlas meshes

Applying RAS to LPS transform to
atlas_FC_inner_mesh_world.ply
atlas_TC_inner_mesh_world.ply
makes them properly aligned with the atlas image.

dzenanz committed:
ENH: Change 3D to 2D projection to work in LPS space

PaulHax committed:
feat: add pipeline_cli.py

dzenanz committed:
Merge pull request #1 from PaulHax/cli

dzenanz committed:
ENH: Fix 3D -> 2D projection

dzenanz committed:
ENH: Use smooth meshes from the atlas

dzenanz committed:
ENH: Use fitted circle's radius to convert from radians to millimeters

dzenanz committed:
ENH: Use smaller circles for projecting vertices to 2D surface

dzenanz committed:
ENH: Match the old orientation of femoral cartilage 2D projection

dzenanz committed:
ENH: Save outer thickness meshes as intermediate results

dzenanz committed:
ENH: New thickness computation approach - version 1

dzenanz committed:
ENH: Use old 3D to 2D projection

dzenanz committed:
ENH: Transform patient distance image to atlas space

This has less artifacts, because the atlas mesh is smooth before transformation.

dzenanz committed:
WIP: Sample from thickness voxels directly

yehan0223 committed:
BUG: Improve mesh splitting

dzenanz committed:
BUG: Do mesh splitting division along Y coordinate

dzenanz committed:
ENH: Simple sampling of thickness via mesh

yehan0223 committed:
ENH: Smoothing before splitting mesh.

yehan0223 committed:
ENH: Keep the largest island(s)

yehan0223 committed:
BUG: Fix TC labels

yehan0223 committed:
ENH: Normalize connection_direction to improve splitting

dzenanz committed:
Trying to fix negative thickness values

dzenanz committed:
ENH: Handle knee laterality by L/R mirroring right knees

We need to do this because the atlas is in left knee orientation.

dzenanz committed:
BUG: Fix 3D to 2D projection having a hole at highest intensity level

dzenanz committed:
Choose mesh splitting or distance-based thickness computation via flag

dzenanz committed:
Reorganize intermediate files

dzenanz committed:
Run all the images from patient 9000798

dzenanz committed:
Move computation specific to image-based thickness into its if branch

PaulHax committed:
ENH: Add laterality arg to pipeline_cli

dzenanz committed:
Use the original (non-smooth) atlas mesh

The 3D to 2D projection results in smoother projection with
the original atlas mesh, which has staircase-like artifacts.

dzenanz committed:
Use 5 instance optimization steps in patient-atlas registration

dzenanz committed:
Use the correct filename for atlas image

dzenanz committed:
Add the missing atlas_image.nii.gz to correct directory

It already existed here: notebooks\OAIData\atlas_image.nii.gz

dzenanz committed:
Remove hard-coded paths from my computer

dzenanz committed:
Reduce instance optimization steps in registration from 5 to 2

dzenanz committed:
Change the test cases to use the currently problematic ones

dzenanz committed:
Avoid holes in 3D -> 2D projection for thickness == 0.0

yehan0223 committed:
ENH: Transform to altas space for better mesh splitting

dzenanz committed:
BUG: Use the exact same image normalization method as before

Originally I used the image normalization method from uniGradICON,
assuming it is the same as what was used before.

dzenanz committed:
Use image plot instead of contour plot when saving to PNG format

dzenanz committed:
ENH: Better island filtering

Keep large islands and islands close to atlas mesh, discard the rest.

dzenanz committed:
ENH: Add the currently problematic cases in local testing

dzenanz committed:
ENH: Use currently problematic cases for testing

dzenanz committed:
ENH: Output mesh before splitting

dzenanz committed:
ENH: Record pip package versions

dzenanz committed:
Update the README.md

* Add pipeline steps
* General update
* Remove old registration approaches

dzenanz committed:
ENH: Update GitHub actions versions in CI

dzenanz committed:
ENH: Use Python 3.11 in CI

dzenanz committed:
ENH: Do not specify torch version in pyproject.toml

This allows CI's torch version to be used.

dzenanz committed:
COMP: Avoid circular import dependency

dzenanz committed:
ENH: Update test_MeshThicknessCPU to comply with thickness refactoring
